library/cartridge_set_control_instance.py

```diff
@@ -15,14 +15,69 @@
 }
 
 
+GET_TWOPHASE_COMMIT_VERSION_TIMEOUT = 10
+
+
+def get_twophase_commit_versions(control_console, advertise_uris):
+    versions, err = control_console.eval_res_err('''
+        local fiber_clock = require('fiber').clock
+        local pool = require('cartridge.pool')
+
+        local uris, timeout = ...
+        local futures = {}
+
+        for i, uri in ipairs(uris) do
+            local conn, err = pool.connect(uri)
```

Comment:
It it possible that this connect will be too long (more than timeout)?

Reply:
It's possible. So "wait_connecter=false" is required here to start "connection" procedure, so then you need to iterate over all connections and call `:wait_connected()`